example_mpi.py

```diff
@@ -0,0 +1,141 @@
+import numpy as np
```

Comment:
again, there seems to be a lot of duplicated code. let's remove duplicate code so there is no confusion about which code is being used.